src/Recoil.rei

```diff
@@ -84,6 +101,16 @@ type value('value) = 'value;
 external useRecoilValue: t('value, 'mode) => value('value) =
   "useRecoilValue";
 
+type loadableValue('a) = {
```

Comment:
Can you also add those? 
https://github.com/facebookexperimental/Recoil/blob/master/src/adt/Recoil_Loadable.js#L44-L51 

They seem more aligned with the language constructs.

Given the polymorphism of the `map` function (we'd need to have `map` and `mapAsync`), I'd make `loadableValue` an abstract type and use externals to access it.

```reason
let loadable = Recoil.useRecoilValueLoadable(atom);

let state = loadable->Recoil.Loadable.state;
let mapped = loadable->Recoil.Loadable.map(value => value * 2);
```